packages/slice-machine/components/ScreenshotChangesModal/Views/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Flex } from "theme-ui";
+import { FiUpload } from "react-icons/fi";
+
+import { ComponentUI, ScreenshotUI } from "@lib/models/common/ComponentUI";
+
+import useCustomScreenshot from "../useCustomScreenshot";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import { useSelector } from "react-redux";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { LoadingKeysEnum } from "@src/modules/loading/types";
+import { isLoading } from "@src/modules/loading";
+
+import EmptyState from "./EmptyState";
+import DefaultView from "./Default";
+
+enum ScreenshotView {
+  Default = 1,
+  EmptyState,
+}
+
+export type ViewRendererProps = {
+  slice: ComponentUI;
+  screenshot: ScreenshotUI;
+  variationID: string;
+  isLoadingScreenshot: boolean;
+};
+
+const viewRenderer = {
+  [ScreenshotView.Default]: DefaultView,
+  [ScreenshotView.EmptyState]: EmptyState,
+};
+
+function VariationScreenshot({
+  variationID,
+  slice,
+}: {
+  variationID: string;
+  slice: ComponentUI;
+}): JSX.Element {
```

Comment:
use `React.FC`, jsx element can be a bit weird :)

Reply:
ahah ok, I suck at this sorry